<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>PhONA-manuscript â€¢ PhONA</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="PhONA-manuscript">
<meta property="og:description" content="PhONA">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-99029062-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-99029062-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">PhONA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/PhONA.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/PhONA-manuscript.html">PhONA-manuscript</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="PhONA-manuscript_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>PhONA-manuscript</h1>
                        <h4 class="author">Ravin Poudel</h4>
            
            <h4 class="date">9/6/2020</h4>
      
      
      <div class="hidden name"><code>PhONA-manuscript.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="co"># GLOBAL VARIABLE</span>
<span class="kw">ITERS</span><span class="op">=</span><span class="fl">5</span>
</pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="co">###### Load the data</span>

<span class="kw">otu_data_fungi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read.mothur.shared.html">read.mothur.shared</a></span>(<span class="st">"data/Fungi1415_trim.contigs.trim.unique.precluster.pick.pick.subsample.nn.unique_list.shared"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">otu_data_fungi</span>) <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"F"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">otu_data_fungi</span>))

<span class="co"># upload meta data file</span>
<span class="kw">bigmetadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span>(<span class="st">"data/block_diversity_selected_tunnel_ww.csv"</span>, header = <span class="kw">T</span>, row.names = <span class="fl">1</span>, stringsAsFactors = <span class="fl">FALSE</span>)
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">bigmetadata</span>) <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"F"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">bigmetadata</span>))

<span class="co">### select otu_data based on metadata</span>
<span class="kw">otu_data</span> <span class="op">&lt;-</span> <span class="kw">otu_data_fungi</span>[<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">bigmetadata</span>), ]
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="kw">otu_data</span>)
</pre></div>
<pre><code>## [1]   160 16151</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="co">#rownames(otu_data)</span>


<span class="kw">otu_data</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html">na.omit</a></span>(<span class="kw">otu_data</span>)

<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="kw">otu_data</span>)
</pre></div>
<pre><code>## [1]   160 16151</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="co"># now select metadata based on od</span>
<span class="kw">meta_data</span> <span class="op">&lt;-</span> <span class="kw">bigmetadata</span>[<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">otu_data</span>), , drop = <span class="fl">FALSE</span>]

<span class="fu"><a href="https://rdrr.io/r/base/all.equal.html">all.equal</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span>(<span class="kw">meta_data</span>), <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span>(<span class="kw">otu_data</span>))
</pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="co"># upload taxanomy file</span>
<span class="kw">tax_fungi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read.mothur.taxonomy.html">read.mothur.taxonomy</a></span>(<span class="st">"data/Fungi1415_trim.contigs.trim.unique.precluster.pick.pick.subsample.nn.unique_list.0.03.cons.taxonomy"</span>)



<span class="co"># checking if taxonomy file and count files have same otus</span>
<span class="fu"><a href="https://rdrr.io/r/base/all.equal.html">all.equal</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="kw">otu_data</span>), <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span>(<span class="kw">tax_fungi</span>))
</pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="co">#</span>
<span class="co">########## read in phenome data/ Yield data</span>
<span class="co"># read in data</span>
<span class="kw">pheno_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span>(<span class="st">"data/tomato_yield.csv"</span>, header = <span class="fl">TRUE</span>, stringsAsFactors = <span class="fl">FALSE</span>, row.names = <span class="fl">1</span>)
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">pheno_data</span>) <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"F"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">pheno_data</span>))

<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span>(<span class="kw">pheno_data</span><span class="op">$</span><span class="kw">Marketable</span>)
</pre></div>
<p><img src="PhONA-manuscript_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="kw">pheno_data_mean</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/aggregate.html">aggregate</a></span>(<span class="kw">pheno_data</span><span class="op">$</span><span class="kw">Marketable</span>, by=<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(Rootstock=<span class="kw">pheno_data</span><span class="op">$</span><span class="kw">Rootstock</span>), FUN=<span class="kw">mean</span>)


<span class="kw">pheno_data_sel</span> <span class="op">&lt;-</span> <span class="kw">pheno_data</span> <span class="op">%&gt;%</span>
  <span class="fu">select</span>(<span class="kw">Rootstock</span>, <span class="kw">Compartment</span>, <span class="kw">Marketable</span>, <span class="kw">Study_site</span>, <span class="kw">Year</span>, <span class="kw">Sample_name</span>)

<span class="kw">meta_withpheo</span> <span class="op">=</span> <span class="fu">inner_join</span>(<span class="kw">meta_data</span>, <span class="kw">pheno_data_sel</span>)
</pre></div>
<pre><code>## Joining, by = c("Rootstock", "Compartment", "Study_site", "Year", "Sample_name")</code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">meta_withpheo</span>)<span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">meta_data</span>)

<span class="co"># create a phyloseq object~ can combine count, metadata, taxonomy, and phylogentic tree.</span>
<span class="kw">tax.mat</span> <span class="op">&lt;-</span> <span class="fu">tax_table</span>(<span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(<span class="kw">tax_fungi</span>))
<span class="kw">otu.mat</span> <span class="op">=</span> <span class="fu">otu_table</span>(<span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="kw">otu_data</span>), taxa_are_rows = <span class="fl">TRUE</span>)
<span class="kw">sample.mat</span> <span class="op">&lt;-</span> <span class="fu">sample_data</span>(<span class="kw">meta_withpheo</span>)
<span class="kw">physeq</span> <span class="op">=</span> <span class="fu">phyloseq</span>(<span class="kw">otu.mat</span>, <span class="kw">tax.mat</span>, <span class="kw">sample.mat</span>)


<span class="co">## Assign color to the taxa on the whole phyloseq object so that the same color is assigned for a taxon across treatments</span>
<span class="kw">physeq</span> <span class="op">=</span> <span class="fu"><a href="../reference/taxacolor.html">taxacolor</a></span>(phyobj = <span class="kw">physeq</span>, coloredby = <span class="st">"Phylum"</span>)

<span class="kw">physeq</span>
</pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 16151 taxa and 160 samples ]
## sample_data() Sample Data:       [ 160 samples by 10 sample variables ]
## tax_table()   Taxonomy Table:    [ 16151 taxa by 9 taxonomic ranks ]</code></pre>
<p>At this point we have a phyloseq object for all the treatments. Now, we will process the phyloseq object the split it by treatment combinations.</p>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="co">#### All the combinations of treatment factor</span>

<span class="kw">region</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="fu">sample_data</span>(<span class="kw">physeq</span>))<span class="op">$</span><span class="kw">Compartment</span>))
<span class="kw">region</span>
</pre></div>
<pre><code>## [1] "Rhizosphere" "Endosphere"</code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="kw">treatment</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="fu">sample_data</span>(<span class="kw">physeq</span>))<span class="op">$</span><span class="kw">Rootstock</span>))
<span class="kw">treatment</span>
</pre></div>
<pre><code>## [1] "Selfgraft"  "RST-04-106" "Nongraft"   "Maxifort"</code></pre>
<p>Getting phyloseq object for endosphere</p>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="co"># Endosphere </span>
<span class="kw">ng_endo</span> <span class="op">&lt;-</span> <span class="fu">subset_samples</span>(<span class="kw">physeq</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">Compartment</span> <span class="op">==</span><span class="st">"Endosphere"</span> <span class="op">&amp;</span> <span class="kw">Rootstock</span><span class="op">==</span><span class="st">"Nongraft"</span>)) <span class="op">%&gt;%</span>
  <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(<span class="kw">.</span>) <span class="op">&gt;</span> <span class="fl">2</span>, <span class="kw">.</span>)
<span class="co"># Inputfile for running SparCC</span>
<span class="kw">ng_endo_sparcc</span> <span class="op">=</span> <span class="fu">otu_table</span>(<span class="kw">ng_endo</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(OTU_id = <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">ng_endo_sparcc</span>), <span class="kw">ng_endo_sparcc</span>), file = <span class="st">"data/sparcc_data/ng_endo_sparcc.txt"</span>, sep = <span class="st">"\t"</span>, row.names = <span class="fl">FALSE</span>, quote = <span class="fl">FALSE</span>)
  


<span class="kw">sg_endo</span> <span class="op">&lt;-</span> <span class="fu">subset_samples</span>(<span class="kw">physeq</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">Compartment</span> <span class="op">==</span><span class="st">"Endosphere"</span> <span class="op">&amp;</span> <span class="kw">Rootstock</span><span class="op">==</span><span class="st">"Selfgraft"</span>)) <span class="op">%&gt;%</span>
    <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(<span class="kw">.</span>) <span class="op">&gt;</span> <span class="fl">2</span>, <span class="kw">.</span>)
<span class="co"># Inputfile for running SparCC</span>
<span class="kw">sg_endo_sparcc</span> <span class="op">=</span> <span class="fu">otu_table</span>(<span class="kw">sg_endo</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(OTU_id = <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">sg_endo_sparcc</span>), <span class="kw">sg_endo_sparcc</span>), file = <span class="st">"data/sparcc_data/sg_endo_sparcc.txt"</span>, sep = <span class="st">"\t"</span>, row.names = <span class="fl">FALSE</span>, quote = <span class="fl">FALSE</span>)
  


<span class="kw">rst_endo</span> <span class="op">&lt;-</span> <span class="fu">subset_samples</span>(<span class="kw">physeq</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">Compartment</span> <span class="op">==</span><span class="st">"Endosphere"</span> <span class="op">&amp;</span> <span class="kw">Rootstock</span><span class="op">==</span><span class="st">"RST-04-106"</span>)) <span class="op">%&gt;%</span>
    <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(<span class="kw">.</span>) <span class="op">&gt;</span> <span class="fl">2</span>, <span class="kw">.</span>)
<span class="co"># Inputfile for running SparCC</span>
<span class="kw">rst_endo_sparcc</span> <span class="op">=</span> <span class="fu">otu_table</span>(<span class="kw">rst_endo</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(OTU_id = <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">rst_endo_sparcc</span>), <span class="kw">rst_endo_sparcc</span>), file = <span class="st">"data/sparcc_data/rst_endo_sparcc.txt"</span>, sep = <span class="st">"\t"</span>, row.names = <span class="fl">FALSE</span>, quote = <span class="fl">FALSE</span>)
  


<span class="kw">maxi_endo</span> <span class="op">&lt;-</span> <span class="fu">subset_samples</span>(<span class="kw">physeq</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">Compartment</span> <span class="op">==</span><span class="st">"Endosphere"</span> <span class="op">&amp;</span> <span class="kw">Rootstock</span><span class="op">==</span><span class="st">"Maxifort"</span>)) <span class="op">%&gt;%</span>
    <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(<span class="kw">.</span>) <span class="op">&gt;</span> <span class="fl">2</span>, <span class="kw">.</span>)
<span class="co"># Inputfile for running SparCC</span>
<span class="kw">maxi_endo_sparcc</span> <span class="op">=</span> <span class="fu">otu_table</span>(<span class="kw">maxi_endo</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(OTU_id = <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">maxi_endo_sparcc</span>), <span class="kw">maxi_endo_sparcc</span>), file = <span class="st">"data/sparcc_data/maxi_endo_sparcc.txt"</span>, sep = <span class="st">"\t"</span>, row.names = <span class="fl">FALSE</span>, quote = <span class="fl">FALSE</span>)
</pre></div>
<p>Getting phyloseq object for rhizosphere</p>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="co"># Rhizosphere</span>

<span class="kw">ng_rhizo</span> <span class="op">&lt;-</span> <span class="fu">subset_samples</span>(<span class="kw">physeq</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">Compartment</span> <span class="op">==</span><span class="st">"Rhizosphere"</span> <span class="op">&amp;</span> <span class="kw">Rootstock</span><span class="op">==</span><span class="st">"Nongraft"</span>)) <span class="op">%&gt;%</span>
    <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(<span class="kw">.</span>) <span class="op">&gt;</span> <span class="fl">2</span>, <span class="kw">.</span>)
<span class="co"># Inputfile for running SparCC</span>
<span class="kw">ng_rhizo_sparcc</span> <span class="op">=</span> <span class="fu">otu_table</span>(<span class="kw">ng_rhizo</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(OTU_id = <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">ng_rhizo_sparcc</span>), <span class="kw">ng_rhizo_sparcc</span>), file = <span class="st">"data/sparcc_data/ng_rhizo_sparcc.txt"</span>, sep = <span class="st">"\t"</span>, row.names = <span class="fl">FALSE</span>, quote = <span class="fl">FALSE</span>)
  

  
<span class="kw">sg_rhizo</span> <span class="op">&lt;-</span> <span class="fu">subset_samples</span>(<span class="kw">physeq</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">Compartment</span> <span class="op">==</span><span class="st">"Rhizosphere"</span> <span class="op">&amp;</span> <span class="kw">Rootstock</span><span class="op">==</span><span class="st">"Selfgraft"</span>)) <span class="op">%&gt;%</span>
    <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(<span class="kw">.</span>) <span class="op">&gt;</span> <span class="fl">2</span>, <span class="kw">.</span>)
<span class="co"># Inputfile for running SparCC</span>
<span class="kw">sg_rhizo_sparcc</span> <span class="op">=</span> <span class="fu">otu_table</span>(<span class="kw">sg_rhizo</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(OTU_id = <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">sg_rhizo_sparcc</span>), <span class="kw">sg_rhizo_sparcc</span>), file = <span class="st">"data/sparcc_data/sg_rhizo_sparcc.txt"</span>, sep = <span class="st">"\t"</span>, row.names = <span class="fl">FALSE</span>, quote = <span class="fl">FALSE</span>)
  

  
<span class="kw">rst_rhizo</span> <span class="op">&lt;-</span> <span class="fu">subset_samples</span>(<span class="kw">physeq</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">Compartment</span> <span class="op">==</span><span class="st">"Rhizosphere"</span> <span class="op">&amp;</span> <span class="kw">Rootstock</span><span class="op">==</span><span class="st">"RST-04-106"</span>))<span class="op">%&gt;%</span>
  <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(<span class="kw">.</span>) <span class="op">&gt;</span> <span class="fl">2</span>, <span class="kw">.</span>)
<span class="co"># Inputfile for running SparCC</span>
<span class="kw">rst_rhizo_sparcc</span> <span class="op">=</span> <span class="fu">otu_table</span>(<span class="kw">rst_rhizo</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(OTU_id = <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">rst_rhizo_sparcc</span>), <span class="kw">rst_rhizo_sparcc</span>), file = <span class="st">"data/sparcc_data/rst_rhizo_sparcc.txt"</span>, sep = <span class="st">"\t"</span>, row.names = <span class="fl">FALSE</span>, quote = <span class="fl">FALSE</span>)
  


<span class="kw">maxi_rhizo</span> <span class="op">&lt;-</span> <span class="fu">subset_samples</span>(<span class="kw">physeq</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">Compartment</span> <span class="op">==</span><span class="st">"Rhizosphere"</span> <span class="op">&amp;</span> <span class="kw">Rootstock</span><span class="op">==</span><span class="st">"Maxifort"</span>)) <span class="op">%&gt;%</span>
  <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(<span class="kw">.</span>) <span class="op">&gt;</span> <span class="fl">2</span>, <span class="kw">.</span>)
<span class="co"># Inputfile for running SparCC</span>
<span class="kw">maxi_rhizo_sparcc</span> <span class="op">=</span> <span class="fu">otu_table</span>(<span class="kw">maxi_rhizo</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(OTU_id = <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">maxi_rhizo_sparcc</span>), <span class="kw">maxi_rhizo_sparcc</span>), file = <span class="st">"data/sparcc_data/maxi_rhizo_sparcc.txt"</span>, sep = <span class="st">"\t"</span>, row.names = <span class="fl">FALSE</span>, quote = <span class="fl">FALSE</span>)
</pre></div>
<p>Read in output from sparcc, then run SparCC.</p>
<p>Nongraft and Endosphere</p>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="kw">ng_endo_sparcc.cor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"data/sparcc_output/Endosphere_Nongraft_cor_sparcc.out"</span>, sep = <span class="st">"\t"</span>, header = <span class="kw">T</span>, row.names = <span class="fl">1</span>)
<span class="kw">ng_endo_sparcc.pval</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"data/sparcc_output/Endosphere_Nongraft_pvals.two_sided.txt"</span>, sep = <span class="st">"\t"</span>, header = <span class="kw">T</span>, row.names = <span class="fl">1</span>)

<span class="kw">phona_ng_endo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PhONA.html">PhONA</a></span>(physeqobj = <span class="kw">ng_endo</span>, cordata = <span class="kw">ng_endo_sparcc.cor</span>,
      pdata = <span class="kw">ng_endo_sparcc.pval</span>, model = <span class="st">"lasso"</span>,
      iters = <span class="kw">ITERS</span>, defineTreatment = <span class="st">"Nongraft"</span>,nodesize = <span class="fl">5</span>,
      PhenoNodesize = <span class="fl">12</span>, definePhenotype = <span class="st">"Marketable"</span>, PhenoNodelabel = <span class="st">"Nongraft"</span>)
</pre></div>
<pre><code>## Total number of iterations used: 5</code></pre>
<p><img src="PhONA-manuscript_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<p>Selfgraft and Endosphere</p>
<div class="sourceCode" id="cb23"><pre class="downlit">
<span class="kw">sg_endo_sparcc.cor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"data/sparcc_output/Endosphere_Selfgraft_cor_sparcc.out"</span>, sep = <span class="st">"\t"</span>, header = <span class="kw">T</span>, row.names = <span class="fl">1</span>)
<span class="kw">sg_endo_sparcc.pval</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"data/sparcc_output/Endosphere_Selfgraft_pvals.two_sided.txt"</span>, sep = <span class="st">"\t"</span>, header = <span class="kw">T</span>, row.names = <span class="fl">1</span>)

<span class="kw">phona_sg_endo</span> <span class="op">&lt;-</span><span class="fu"><a href="../reference/PhONA.html">PhONA</a></span>(physeqobj = <span class="kw">sg_endo</span>, cordata = <span class="kw">sg_endo_sparcc.cor</span>,
      pdata = <span class="kw">sg_endo_sparcc.pval</span>, model = <span class="st">"lasso"</span>,
      iters = <span class="kw">ITERS</span>, defineTreatment = <span class="st">"Selfgraft"</span>,nodesize = <span class="fl">5</span>,
      PhenoNodesize = <span class="fl">12</span>, definePhenotype = <span class="st">"Marketable"</span>, PhenoNodelabel = <span class="st">"Selfgraft"</span>)
</pre></div>
<pre><code>## Total number of iterations used: 5</code></pre>
<p><img src="PhONA-manuscript_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p>RST-04-106 and Endosphere</p>
<div class="sourceCode" id="cb25"><pre class="downlit">
<span class="kw">rst_endo_sparcc.cor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"data/sparcc_output/Endosphere_RST-04-106_cor_sparcc.out"</span>, sep = <span class="st">"\t"</span>, header = <span class="kw">T</span>, row.names = <span class="fl">1</span>)
<span class="kw">rst_endo_sparcc.pval</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"data/sparcc_output/Endosphere_RST-04-106_pvals.two_sided.txt"</span>, sep = <span class="st">"\t"</span>, header = <span class="kw">T</span>, row.names = <span class="fl">1</span>)

<span class="kw">phona_rst_endo</span> <span class="op">&lt;-</span><span class="fu"><a href="../reference/PhONA.html">PhONA</a></span>(physeqobj = <span class="kw">rst_endo</span>, cordata = <span class="kw">rst_endo_sparcc.cor</span>,
      pdata = <span class="kw">rst_endo_sparcc.pval</span>, model = <span class="st">"lasso"</span>,
      iters = <span class="kw">ITERS</span>, defineTreatment = <span class="st">"RST-04-106"</span>,nodesize = <span class="fl">5</span>,
      PhenoNodesize = <span class="fl">12</span>, definePhenotype = <span class="st">"Marketable"</span>, PhenoNodelabel = <span class="st">"RST-04-106"</span>)
</pre></div>
<pre><code>## Total number of iterations used: 5</code></pre>
<p><img src="PhONA-manuscript_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>Maxifort and Endosphere</p>
<div class="sourceCode" id="cb27"><pre class="downlit">
<span class="kw">maxi_endo_sparcc.cor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"data/sparcc_output/Endosphere_Maxifort_cor_sparcc.out"</span>, sep = <span class="st">"\t"</span>, header = <span class="kw">T</span>, row.names = <span class="fl">1</span>)
<span class="kw">maxi_endo_sparcc.pval</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"data/sparcc_output/Endosphere_Maxifort_pvals.two_sided.txt"</span>, sep = <span class="st">"\t"</span>, header = <span class="kw">T</span>, row.names = <span class="fl">1</span>)

<span class="kw">phona_maxi_endo</span> <span class="op">&lt;-</span><span class="fu"><a href="../reference/PhONA.html">PhONA</a></span>(physeqobj = <span class="kw">maxi_endo</span>, cordata = <span class="kw">maxi_endo_sparcc.cor</span>,
      pdata = <span class="kw">maxi_endo_sparcc.pval</span>, model = <span class="st">"lasso"</span>,
      iters = <span class="kw">ITERS</span>, defineTreatment = <span class="st">"Maxifort"</span>,nodesize = <span class="fl">5</span>,
      PhenoNodesize = <span class="fl">12</span>, definePhenotype = <span class="st">"Marketable"</span>, PhenoNodelabel = <span class="st">"Maxifort"</span>)
</pre></div>
<pre><code>## Total number of iterations used: 5</code></pre>
<p><img src="PhONA-manuscript_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<p>Nongraft and Rhizosphere</p>
<div class="sourceCode" id="cb29"><pre class="downlit">
<span class="kw">ng_rhizo_sparcc.cor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"data/sparcc_output/Rhizosphere_Nongraft_cor_sparcc.out"</span>, sep = <span class="st">"\t"</span>, header = <span class="kw">T</span>, row.names = <span class="fl">1</span>)
<span class="kw">ng_rhizo_sparcc.pval</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"data/sparcc_output/Rhizosphere_Nongraft_pvals.two_sided.txt"</span>, sep = <span class="st">"\t"</span>, header = <span class="kw">T</span>, row.names = <span class="fl">1</span>)

<span class="kw">phona_ng_rhizo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PhONA.html">PhONA</a></span>(physeqobj = <span class="kw">ng_rhizo</span>, cordata = <span class="kw">ng_rhizo_sparcc.cor</span>,
      pdata = <span class="kw">ng_rhizo_sparcc.pval</span>, model = <span class="st">"lasso"</span>,
      iters = <span class="kw">ITERS</span>, defineTreatment = <span class="st">"Nongraft"</span>,nodesize = <span class="fl">5</span>,
      PhenoNodesize = <span class="fl">12</span>, definePhenotype = <span class="st">"Marketable"</span>, PhenoNodelabel = <span class="st">"Nongraft"</span>)
</pre></div>
<pre><code>## Total number of iterations used: 5</code></pre>
<p><img src="PhONA-manuscript_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p>Selfgraft and Rhizosphere</p>
<div class="sourceCode" id="cb31"><pre class="downlit">
<span class="kw">sg_rhizo_sparcc.cor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"data/sparcc_output/Rhizosphere_Selfgraft_cor_sparcc.out"</span>, sep = <span class="st">"\t"</span>, header = <span class="kw">T</span>, row.names = <span class="fl">1</span>)
<span class="kw">sg_rhizo_sparcc.pval</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"data/sparcc_output/Rhizosphere_Selfgraft_pvals.two_sided.txt"</span>, sep = <span class="st">"\t"</span>, header = <span class="kw">T</span>, row.names = <span class="fl">1</span>)

<span class="kw">phona_sg_rhizo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PhONA.html">PhONA</a></span>(physeqobj = <span class="kw">sg_rhizo</span>, cordata = <span class="kw">sg_rhizo_sparcc.cor</span>,
      pdata = <span class="kw">sg_rhizo_sparcc.pval</span>, model = <span class="st">"lasso"</span>,
      iters = <span class="kw">ITERS</span>, defineTreatment = <span class="st">"Selfgraft"</span>,nodesize = <span class="fl">5</span>,
      PhenoNodesize = <span class="fl">12</span>, definePhenotype = <span class="st">"Marketable"</span>, PhenoNodelabel = <span class="st">"Selfgraft"</span>)
</pre></div>
<pre><code>## Total number of iterations used: 5</code></pre>
<p><img src="PhONA-manuscript_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<p>RST-04-106 and Rhizosphere</p>
<div class="sourceCode" id="cb33"><pre class="downlit">
<span class="kw">rst_rhizo_sparcc.cor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"data/sparcc_output/Rhizosphere_RST-04-106_cor_sparcc.out"</span>, sep = <span class="st">"\t"</span>, header = <span class="kw">T</span>, row.names = <span class="fl">1</span>)
<span class="kw">rst_rhizo_sparcc.pval</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"data/sparcc_output/Rhizosphere_RST-04-106_pvals.two_sided.txt"</span>, sep = <span class="st">"\t"</span>, header = <span class="kw">T</span>, row.names = <span class="fl">1</span>)

<span class="kw">phona_rst_rhizo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PhONA.html">PhONA</a></span>(physeqobj = <span class="kw">rst_rhizo</span>, cordata = <span class="kw">rst_rhizo_sparcc.cor</span>,
      pdata = <span class="kw">rst_rhizo_sparcc.pval</span>, model = <span class="st">"lasso"</span>,
      iters = <span class="kw">ITERS</span>, defineTreatment = <span class="st">"RST-04-106"</span>,nodesize = <span class="fl">5</span>,
      PhenoNodesize = <span class="fl">12</span>, definePhenotype = <span class="st">"Marketable"</span>, PhenoNodelabel = <span class="st">"RST-04-106"</span>)
</pre></div>
<pre><code>## Total number of iterations used: 5</code></pre>
<p><img src="PhONA-manuscript_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>Maxifort and Rhizosphere</p>
<div class="sourceCode" id="cb35"><pre class="downlit">
<span class="kw">maxi_rhizo_sparcc.cor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"data/sparcc_output/Rhizosphere_Maxifort_cor_sparcc.out"</span>, sep = <span class="st">"\t"</span>, header = <span class="kw">T</span>, row.names = <span class="fl">1</span>)
<span class="kw">maxi_rhizo_sparcc.pval</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"data/sparcc_output/Rhizosphere_Maxifort_pvals.two_sided.txt"</span>, sep = <span class="st">"\t"</span>, header = <span class="kw">T</span>, row.names = <span class="fl">1</span>)

<span class="kw">phona_maxi_rhizo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PhONA.html">PhONA</a></span>(physeqobj = <span class="kw">maxi_rhizo</span>, cordata = <span class="kw">maxi_rhizo_sparcc.cor</span>,
      pdata = <span class="kw">maxi_rhizo_sparcc.pval</span>, model = <span class="st">"lasso"</span>,
      defineTreatment = <span class="st">"Maxifort"</span>,nodesize = <span class="fl">5</span>,
      iters = <span class="kw">ITERS</span>, PhenoNodesize = <span class="fl">12</span>, definePhenotype = <span class="st">"Marketable"</span>, PhenoNodelabel = <span class="st">"Maxifort"</span>)
</pre></div>
<pre><code>## Total number of iterations used: 5</code></pre>
<p><img src="PhONA-manuscript_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<p>Role analyses using SA algorithm as implemented in rnetcarto package.</p>
<p>Endosphere</p>
<div class="sourceCode" id="cb37"><pre class="downlit">
<span class="co"># Endosphere </span>

<span class="kw">role_df_Endosphere</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span>(<span class="kw">phona_ng_endo</span><span class="op">$</span><span class="kw">roles</span>,
                            <span class="kw">phona_sg_endo</span><span class="op">$</span><span class="kw">roles</span>,
                            <span class="kw">phona_rst_endo</span><span class="op">$</span><span class="kw">roles</span>,
                            <span class="kw">phona_maxi_endo</span><span class="op">$</span><span class="kw">roles</span>)
</pre></div>
<div class="sourceCode" id="cb38"><pre class="downlit">
<span class="kw">fig3b_endo</span> <span class="op">&lt;-</span><span class="fu">ggplot</span>(<span class="kw">role_df_Endosphere</span>, <span class="fu">aes</span>(<span class="kw">participation</span>,<span class="kw">connectivity</span>,label=<span class="kw">name</span>)) <span class="op">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(colour = <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="kw">Treatment</span>)))<span class="op">+</span>
  <span class="fu">scale_size</span>(guide = <span class="st">'none'</span>)<span class="op">+</span>
  <span class="fu">scale_color_manual</span>(breaks= <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Nongraft"</span>, <span class="st">"Selfgraft"</span>, <span class="st">"RST-04-106"</span>, <span class="st">"Maxifort"</span>),values = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"mediumaquamarine"</span>,<span class="st">"sienna4"</span>,<span class="st">"#ffae19"</span>,<span class="st">"blueviolet"</span>))<span class="op">+</span>
  <span class="fu">geom_text</span>(<span class="fu">aes</span>(label=<span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="kw">role</span>!=<span class="st">"Peripheral"</span>,<span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="kw">Order</span>),<span class="st">''</span>)),hjust=<span class="fl">0.1</span>,vjust=<span class="fl">0.1</span>,size=<span class="fl">3</span>,color=<span class="st">"black"</span>,family=<span class="st">"Times New Roman"</span>)<span class="op">+</span>
  <span class="fu">geom_hline</span>(yintercept=<span class="fl">2.5</span>, linetype=<span class="st">"dashed"</span>, color = <span class="st">"red"</span>)<span class="op">+</span>
  <span class="fu">geom_vline</span>(xintercept = <span class="fl">0.62</span>, linetype=<span class="st">"dashed"</span>,color = <span class="st">"blue"</span>)<span class="op">+</span>
  <span class="fu">annotate</span>(<span class="st">"text"</span>, x = <span class="fl">0.2</span>, y = <span class="fl">3</span>, label = <span class="st">"Module hubs"</span>,color=<span class="st">"gray20"</span>,size=<span class="fl">8</span>,family=<span class="st">"Times New Roman"</span>)<span class="op">+</span>
  <span class="fu">annotate</span>(<span class="st">"text"</span>, x = <span class="fl">0.75</span>, y = <span class="fl">3</span>, label = <span class="st">" Network hubs"</span>,color=<span class="st">"gray20"</span>,size=<span class="fl">8</span>,family=<span class="st">"Times New Roman"</span>)<span class="op">+</span>
  <span class="fu">annotate</span>(<span class="st">"text"</span>, x = <span class="fl">0.2</span>, y = <span class="op">-</span><span class="fl">2</span>, label = <span class="st">"Peripherals"</span>,color=<span class="st">"gray20"</span>,size=<span class="fl">8</span>,family=<span class="st">"Times New Roman"</span>)<span class="op">+</span>
  <span class="fu">annotate</span>(<span class="st">"text"</span>, x = <span class="fl">0.75</span>, y = <span class="op">-</span><span class="fl">2</span>, label = <span class="st">"Connectors"</span>,color=<span class="st">"gray20"</span>,size=<span class="fl">8</span>,family=<span class="st">"Times New Roman"</span>)<span class="op">+</span>
  <span class="fu">labs</span>(x = <span class="st">"Among-module connectivity"</span>, y = <span class="st">"Within-module degree"</span>, col=<span class="st">"Rootstocks"</span>) <span class="op">+</span>
  <span class="fu">theme_bw</span>() <span class="op">+</span>
  <span class="fu">scale_x_continuous</span>(limits = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">1.0</span>))<span class="op">+</span>
  <span class="fu">theme</span>(axis.text=<span class="fu">element_text</span>(size=<span class="fl">16</span>,face=<span class="st">"bold"</span>,family=<span class="st">"Times New Roman"</span>),axis.title=<span class="fu">element_text</span>(size=<span class="fl">16</span>,face=<span class="st">"bold"</span>,family=<span class="st">"Times New Roman"</span>))<span class="op">+</span>
  <span class="fu">labs</span>(fill = <span class="st">"Dose (mg)"</span>) <span class="op">+</span>
  <span class="fu">theme</span>(legend.title=<span class="fu">element_text</span>(size=<span class="fl">16</span>,face=<span class="st">"bold"</span>,family=<span class="st">"Times New Roman"</span>),
        legend.text=<span class="fu">element_text</span>(size=<span class="fl">16</span>,family=<span class="st">"Times New Roman"</span>)) <span class="op">+</span>
  <span class="fu">guides</span>(color = <span class="fu">guide_legend</span>(override.aes = <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(size=<span class="fl">5</span>)))

<span class="kw">fig3b_endo</span>
</pre></div>
<p><img src="PhONA-manuscript_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<p>Rhizosphere</p>
<div class="sourceCode" id="cb39"><pre class="downlit">
<span class="co"># Endosphere </span>

<span class="kw">role_df_Rhizosphere</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span>(<span class="kw">phona_ng_rhizo</span><span class="op">$</span><span class="kw">roles</span>,
                             <span class="kw">phona_sg_rhizo</span><span class="op">$</span><span class="kw">roles</span>,
                             <span class="kw">phona_rst_rhizo</span><span class="op">$</span><span class="kw">roles</span>,
                             <span class="kw">phona_maxi_rhizo</span><span class="op">$</span><span class="kw">roles</span>)
</pre></div>
<div class="sourceCode" id="cb40"><pre class="downlit">
<span class="kw">fig3b_rhizo</span> <span class="op">&lt;-</span><span class="fu">ggplot</span>(<span class="kw">role_df_Rhizosphere</span>, <span class="fu">aes</span>(<span class="kw">participation</span>,<span class="kw">connectivity</span>,label=<span class="kw">name</span>)) <span class="op">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(colour = <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="kw">Treatment</span>)))<span class="op">+</span>
  <span class="fu">scale_size</span>(guide = <span class="st">'none'</span>)<span class="op">+</span>
  <span class="fu">scale_color_manual</span>(breaks= <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Nongraft"</span>, <span class="st">"Selfgraft"</span>, <span class="st">"RST-04-106"</span>, <span class="st">"Maxifort"</span>),values = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"mediumaquamarine"</span>,<span class="st">"sienna4"</span>,<span class="st">"#ffae19"</span>,<span class="st">"blueviolet"</span>))<span class="op">+</span>
  <span class="fu">geom_text</span>(<span class="fu">aes</span>(label=<span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="kw">role</span>!=<span class="st">"Peripheral"</span>,<span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="kw">Order</span>),<span class="st">''</span>)),hjust=<span class="fl">0.1</span>,vjust=<span class="fl">0.1</span>,size=<span class="fl">3</span>,color=<span class="st">"black"</span>,family=<span class="st">"Times New Roman"</span>)<span class="op">+</span>
  <span class="fu">geom_hline</span>(yintercept=<span class="fl">2.5</span>, linetype=<span class="st">"dashed"</span>, color = <span class="st">"red"</span>)<span class="op">+</span>
  <span class="fu">geom_vline</span>(xintercept = <span class="fl">0.62</span>, linetype=<span class="st">"dashed"</span>,color = <span class="st">"blue"</span>)<span class="op">+</span>
  <span class="fu">annotate</span>(<span class="st">"text"</span>, x = <span class="fl">0.2</span>, y = <span class="fl">3</span>, label = <span class="st">"Module hubs"</span>,color=<span class="st">"gray20"</span>,size=<span class="fl">8</span>,family=<span class="st">"Times New Roman"</span>)<span class="op">+</span>
  <span class="fu">annotate</span>(<span class="st">"text"</span>, x = <span class="fl">0.75</span>, y = <span class="fl">3</span>, label = <span class="st">" Network hubs"</span>,color=<span class="st">"gray20"</span>,size=<span class="fl">8</span>,family=<span class="st">"Times New Roman"</span>)<span class="op">+</span>
  <span class="fu">annotate</span>(<span class="st">"text"</span>, x = <span class="fl">0.2</span>, y = <span class="op">-</span><span class="fl">2</span>, label = <span class="st">"Peripherals"</span>,color=<span class="st">"gray20"</span>,size=<span class="fl">8</span>,family=<span class="st">"Times New Roman"</span>)<span class="op">+</span>
  <span class="fu">annotate</span>(<span class="st">"text"</span>, x = <span class="fl">0.75</span>, y = <span class="op">-</span><span class="fl">2</span>, label = <span class="st">"Connectors"</span>,color=<span class="st">"gray20"</span>,size=<span class="fl">8</span>,family=<span class="st">"Times New Roman"</span>)<span class="op">+</span>
  <span class="fu">labs</span>(x = <span class="st">"Among-module connectivity"</span>, y = <span class="st">"Within-module degree"</span>, col=<span class="st">"Rootstocks"</span>) <span class="op">+</span>
  <span class="fu">theme_bw</span>() <span class="op">+</span>
  <span class="fu">scale_x_continuous</span>(limits = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">1.0</span>))<span class="op">+</span>
  <span class="fu">theme</span>(axis.text=<span class="fu">element_text</span>(size=<span class="fl">16</span>,face=<span class="st">"bold"</span>,family=<span class="st">"Times New Roman"</span>),axis.title=<span class="fu">element_text</span>(size=<span class="fl">16</span>,face=<span class="st">"bold"</span>,family=<span class="st">"Times New Roman"</span>))<span class="op">+</span>
  <span class="fu">labs</span>(fill = <span class="st">"Dose (mg)"</span>) <span class="op">+</span>
  <span class="fu">theme</span>(legend.title=<span class="fu">element_text</span>(size=<span class="fl">16</span>,face=<span class="st">"bold"</span>,family=<span class="st">"Times New Roman"</span>),
        legend.text=<span class="fu">element_text</span>(size=<span class="fl">16</span>,family=<span class="st">"Times New Roman"</span>)) <span class="op">+</span>
  <span class="fu">guides</span>(color = <span class="fu">guide_legend</span>(override.aes = <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(size=<span class="fl">5</span>)))

<span class="kw">fig3b_rhizo</span>
</pre></div>
<p><img src="PhONA-manuscript_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
<p>Summary of all the graphs:</p>
<p>Endosphere</p>
<div class="sourceCode" id="cb41"><pre class="downlit">
<span class="kw">summary_graph_endo_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span>(<span class="kw">phona_ng_endo</span><span class="op">$</span><span class="kw">graph_summary</span>,
                          <span class="kw">phona_sg_endo</span><span class="op">$</span><span class="kw">graph_summary</span>,
                          <span class="kw">phona_rst_endo</span><span class="op">$</span><span class="kw">graph_summary</span>,
                          <span class="kw">phona_maxi_endo</span><span class="op">$</span><span class="kw">graph_summary</span>)
<span class="fu">kable</span>(<span class="kw">summary_graph_endo_df</span>)
</pre></div>
<table class="table">
<colgroup>
<col width="3%">
<col width="3%">
<col width="8%">
<col width="6%">
<col width="4%">
<col width="4%">
<col width="9%">
<col width="3%">
<col width="9%">
<col width="7%">
<col width="11%">
<col width="8%">
<col width="9%">
<col width="8%">
</colgroup>
<thead><tr class="header">
<th align="right">node</th>
<th align="right">edge</th>
<th align="right">nodeDegree</th>
<th align="right">avgpath</th>
<th align="right">trans</th>
<th align="right">mod</th>
<th align="right">connectance</th>
<th align="right">wtc</th>
<th align="right">nModules.SA</th>
<th align="left">top3hub</th>
<th align="left">top3hubv</th>
<th align="right">n_postiveL</th>
<th align="right">n_negativeL</th>
<th align="left">Treatment</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">77</td>
<td align="right">120</td>
<td align="right">1.558</td>
<td align="right">5.774</td>
<td align="right">0.559</td>
<td align="right">0.751</td>
<td align="right">0.020</td>
<td align="right">13</td>
<td align="right">9</td>
<td align="left">54;5;64</td>
<td align="left">1;0.978;0.915</td>
<td align="right">111</td>
<td align="right">5</td>
<td align="left">Nongraft</td>
</tr>
<tr class="even">
<td align="right">81</td>
<td align="right">111</td>
<td align="right">1.370</td>
<td align="right">6.380</td>
<td align="right">0.417</td>
<td align="right">0.754</td>
<td align="right">0.017</td>
<td align="right">15</td>
<td align="right">11</td>
<td align="left">23;41;14</td>
<td align="left">1;0.963;0.72</td>
<td align="right">97</td>
<td align="right">9</td>
<td align="left">Selfgraft</td>
</tr>
<tr class="odd">
<td align="right">70</td>
<td align="right">107</td>
<td align="right">1.529</td>
<td align="right">4.225</td>
<td align="right">0.558</td>
<td align="right">0.763</td>
<td align="right">0.022</td>
<td align="right">13</td>
<td align="right">11</td>
<td align="left">19;48;54</td>
<td align="left">1;0.904;0.904</td>
<td align="right">95</td>
<td align="right">9</td>
<td align="left">RST-04-106</td>
</tr>
<tr class="even">
<td align="right">92</td>
<td align="right">184</td>
<td align="right">2.000</td>
<td align="right">3.716</td>
<td align="right">0.596</td>
<td align="right">0.615</td>
<td align="right">0.022</td>
<td align="right">14</td>
<td align="right">10</td>
<td align="left">36;2;11</td>
<td align="left">1;0.925;0.885</td>
<td align="right">148</td>
<td align="right">32</td>
<td align="left">Maxifort</td>
</tr>
</tbody>
</table>
<p>Rhizosphere</p>
<div class="sourceCode" id="cb42"><pre class="downlit">
<span class="kw">summary_graph_rhizo_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span>(<span class="kw">phona_ng_rhizo</span><span class="op">$</span><span class="kw">graph_summary</span>,
                          <span class="kw">phona_sg_rhizo</span><span class="op">$</span><span class="kw">graph_summary</span>,
                          <span class="kw">phona_rst_rhizo</span><span class="op">$</span><span class="kw">graph_summary</span>,
                          <span class="kw">phona_maxi_rhizo</span><span class="op">$</span><span class="kw">graph_summary</span>)
<span class="fu">kable</span>(<span class="kw">summary_graph_rhizo_df</span>)
</pre></div>
<table class="table">
<colgroup>
<col width="3%">
<col width="3%">
<col width="8%">
<col width="6%">
<col width="4%">
<col width="4%">
<col width="9%">
<col width="3%">
<col width="9%">
<col width="7%">
<col width="11%">
<col width="8%">
<col width="9%">
<col width="8%">
</colgroup>
<thead><tr class="header">
<th align="right">node</th>
<th align="right">edge</th>
<th align="right">nodeDegree</th>
<th align="right">avgpath</th>
<th align="right">trans</th>
<th align="right">mod</th>
<th align="right">connectance</th>
<th align="right">wtc</th>
<th align="right">nModules.SA</th>
<th align="left">top3hub</th>
<th align="left">top3hubv</th>
<th align="right">n_postiveL</th>
<th align="right">n_negativeL</th>
<th align="left">Treatment</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">132</td>
<td align="right">336</td>
<td align="right">2.545</td>
<td align="right">4.078</td>
<td align="right">0.508</td>
<td align="right">0.597</td>
<td align="right">0.019</td>
<td align="right">15</td>
<td align="right">12</td>
<td align="left">17;39;36</td>
<td align="left">1;0.866;0.761</td>
<td align="right">257</td>
<td align="right">72</td>
<td align="left">Nongraft</td>
</tr>
<tr class="even">
<td align="right">112</td>
<td align="right">268</td>
<td align="right">2.393</td>
<td align="right">3.892</td>
<td align="right">0.471</td>
<td align="right">0.591</td>
<td align="right">0.021</td>
<td align="right">16</td>
<td align="right">8</td>
<td align="left">20;10;11</td>
<td align="left">1;0.961;0.767</td>
<td align="right">200</td>
<td align="right">62</td>
<td align="left">Selfgraft</td>
</tr>
<tr class="odd">
<td align="right">133</td>
<td align="right">336</td>
<td align="right">2.526</td>
<td align="right">4.215</td>
<td align="right">0.497</td>
<td align="right">0.645</td>
<td align="right">0.019</td>
<td align="right">17</td>
<td align="right">12</td>
<td align="left">39;67;27</td>
<td align="left">1;0.978;0.811</td>
<td align="right">280</td>
<td align="right">54</td>
<td align="left">RST-04-106</td>
</tr>
<tr class="even">
<td align="right">172</td>
<td align="right">736</td>
<td align="right">4.279</td>
<td align="right">3.078</td>
<td align="right">0.594</td>
<td align="right">0.379</td>
<td align="right">0.025</td>
<td align="right">21</td>
<td align="right">11</td>
<td align="left">22;39;73</td>
<td align="left">1;0.954;0.914</td>
<td align="right">451</td>
<td align="right">278</td>
<td align="left">Maxifort</td>
</tr>
</tbody>
</table>
</div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Ravin Poudel.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
